### ISSUE TITLE
Text selection, intuitive cursor position?

### ISSUE THREAD

**Issue by tamo** [geometer/FBReaderJ#237]
Though it depends on indivisual user's preference, the current cursor behavior looks unintuitive
when you long-press to select a text.

1) The initial position should not be adjusted, IMO. I usually press the middle of the height, not the bottom of the text.
```diff
--- a/src/org/geometerplus/zlibrary/text/view/ZLTextView.java
+++ b/src/org/geometerplus/zlibrary/text/view/ZLTextView.java
@@ -1685,7 +1685,7 @@ public abstract class ZLTextView extends ZLTextViewBase {
	}
 
 	protected boolean initSelection(int x, int y) {
-		y -= ZLTextSelectionCursor.getHeight() / 2 + ZLTextSelectionCursor.getAccent() / 2;
+		//y -= ZLTextSelectionCursor.getHeight() / 2 + ZLTextSelectionCursor.getAccent() / 2;
 		if (!mySelection.start(x, y)) {
 			return false;
 		}
```

2) The cursor's head should be the same as the cursor's real position. Currently the cursor is a little confusing.
```diff
@@ -432,7 +432,7 @@ private void drawSelectionCursor(ZLPaintContext context, ZLTextSelection.Point pt) {
 		final int h = ZLTextSelectionCursor.getHeight();
 		final int a = ZLTextSelectionCursor.getAccent();
 		final int[] xs = { pt.X, pt.X + w, pt.X + w, pt.X - w, pt.X - w };
-		final int[] ys = { pt.Y - a, pt.Y, pt.Y + h, pt.Y + h, pt.Y };
+		final int[] ys = { pt.Y, pt.Y + a, pt.Y + h, pt.Y + h, pt.Y + a };
 		context.setFillColor(context.getBackgroundColor(), 192);
 		context.fillPolygon(xs, ys);
 		context.setLineColor(getTextColor(ZLTextHyperlink.NO_LINK));
```

3) I'm not sure whether the moving cursor should be adjusted or not. Some may want, others may not want.
```diff
@@ -326,7 +326,7 @@ public abstract class ZLTextView extends ZLTextViewBase {
 	}
 
 	protected void moveSelectionCursorTo(ZLTextSelectionCursor cursor, int x, int y) {
-		y -= ZLTextSelectionCursor.getHeight() / 2 + ZLTextSelectionCursor.getAccent() / 2;
+		//y -= ZLTextSelectionCursor.getHeight() / 2 + ZLTextSelectionCursor.getAccent() / 2;
 		mySelection.setCursorInMovement(cursor, x, y);
 		mySelection.expandTo(myCurrentPage, x, y);
 		Application.getViewWidget().reset();
```
(also ZLTextSelextion.expandTo needs modifying if you decide to delete the adjustment)

4) Cursor moves slowly.
It would be useful if I could specify the end of selection with a tap.
i.e. Long tap -> Release -> Pop-up -> Choose "Re-select by tapping the right-edge of the selection" (new icon, maybe with an image of finger) -> Tap -> expandTo() -> Pop-up

You may want to make new options for 1, 2 and 3 since each user has different size of fingers or thumbs.

**Comment by tamo** [comment]
Now it's fixed, isn't it?
Maybe fixed with 336821c520b2498c9e7985aa032370281ae9c9af

**Comment by geometer** [comment]
Yes, it is fixed, thanks!
